framework/src/st_things/things_stack/utils/things_network.c

```diff
@@ -380,12 +381,42 @@ void things_wifi_scan_done(wifi_manager_scan_info_s **scan_result, int res)
 	access_point_info_s *pinfo = NULL;
 	access_point_info_s *p_last_info = NULL;
 	while (wifi_scan_iter != NULL) {
-		if (wifi_scan_iter->ssid != NULL) {
+		if (strlen(wifi_scan_iter->ssid) != 0) {
 			pinfo = (access_point_info_s*)things_malloc(sizeof(access_point_info_s));
 			pinfo->next = NULL;
 			snprintf(pinfo->e_ssid, WIFIMGR_SSID_LEN, "%s", wifi_scan_iter->ssid);
 			snprintf(pinfo->bss_id, WIFIMGR_MACADDR_STR_LEN, "%s", wifi_scan_iter->bssid);
 			snprintf(pinfo->signal_level, MAX_LEVEL_SIGNAL, "%d", wifi_scan_iter->rssi);
+			//set auth type
+			const char *sec_type;
+			if (wifi_scan_iter->ap_auth_type == WIFI_MANAGER_AUTH_WEP_SHARED) {
+				sec_type = "WEP-SH";
```

Comment:
Define these constant strings("WEP", "WPA-PSK") in things_def.h as MACRO and use it here.

Reply:
done.